ludwig/utils/llm_utils.py

```diff
@@ -1,14 +1,17 @@
 import copy
 import logging
-from typing import Dict, Tuple
+import tempfile
+from typing import Dict, Tuple, Union
 
 import torch
 import torch.nn.functional as F
 from bitsandbytes.nn.modules import Embedding
-from transformers import AutoConfig, AutoModelForCausalLM, PreTrainedTokenizer
+from peft import get_peft_model, MODEL_TYPE_TO_PEFT_MODEL_MAPPING, PeftConfig, PeftModel, TaskType
```

Comment:
@jeffkinnison seems like tests are failing due to a peft import failure. Can you wrap this in try/except logic like we do here? https://github.com/ludwig-ai/ludwig/blob/8f10afe609fd10fb08afcb6063a9dfb8ad7f41e9/ludwig/utils/visualization_utils.py#L31-L52

Reply:
@jeffkinnison Another option is to just import these functions inline - that way they only get imported when the method is called, which is fine because it means that the user has the llm dependencies already installed (I believe that is what we were doing before)

Reply:
Done, I reverted it to the inline format.